certbot-nginx/tests/configurator_test.py

```diff
@@ -192,8 +191,9 @@ def test_choose_vhosts_bad(self):
                        '69.255.225.155']
 
         for name in bad_results:
-            self.assertRaises(errors.MisconfigurationError,
-                              self.config.choose_vhosts, name)
+            with self.subTest(name=name):
```

Comment:
What is the purpose of this addition?

Reply:
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests When one of them fails, you don't get feedback about the rest. I tend to prefer this when there are `for`-loops in the code. It might fall a bit of scope of this PR which was mainly to use more expressive assertions.

Reply:
Thanks, I was not aware of this possibility!